staker/README.md

```diff
@@ -0,0 +1,170 @@
+```math
+\text{poolReward}(\mathrm{pool}) 
+= \frac{\text{emission} \,\times\, \mathrm{TierRatio}\!\bigl(\mathrm{tier}(\mathrm{pool})\bigr)}
+       {\mathrm{Count}\!\bigl(\mathrm{tier}(\mathrm{pool})\bigr)},
+```
+
+```math
+\mathrm{TierRatio}(t) \;=\;
+\begin{cases}
+[1,\,0,\,0]_{\,t-1}, 
+& \text{if } \mathrm{Count}(2) = 0 \;\land\; \mathrm{Count}(3) = 0, \\[8pt]
+[0.8,\,0,\,0.2]_{\,t-1}, 
+& \text{if } \mathrm{Count}(2) = 0, \\[8pt]
+[0.7,\,0.3,\,0]_{\,t-1}, 
+& \text{if } \mathrm{Count}(3) = 0, \\[8pt]
+[0.5,\,0.3,\,0.2]_{\,t-1}, 
+& \text{otherwise}.
+\end{cases}
+```
+
+```math
+\text{emission} 
+= \mathrm{GNSEmissionPerSecond} 
+  \;\times\;
+  \biggl(\frac{\mathrm{avgMsPerBlock}}{1000}\biggr)
+  \;\times\;
+  \mathrm{StakerEmissionRatio}.
+```
+
+There is always at least one tier-1 pool.
+
+The result of this equation could be changed when variable emission or Count(t) changes.
+
+Emission(for staker contract) is calculated by applying avgMsPerBlock and StakerEmissionRatio to GNSEmissionPerSecond. GNSEmissionPerSecond is constant(other than halving event, which we will ignore in this paragraph). When one of those two variables are modified, the `callbackStakerEmissionChange` function is called, which will caches the reward until the current block and updates the current emission with the provided amount.
+
+PoolTier.cacheReward recalculates all reward-related updates happened since the last cache height.
+- Halving blocks: if there were halving events, split caching operation at the halving periods. The caching will be done for each halving periods, staker emission is updated, and then the caching operation moves forward.
+- Unclaimable period: if the pool is currently in the unclaimable period(there are no claimable stakers), it will updates the unclaimable accumulation using the old emission, and then updates to start a new period(only for the internal)
+- After checking these, it updates GlobalRewardRatioAccumulation, which is used to calculate the rewards for the positions.
+
+Once the cache is filled until the current block, CalculateInternalReward is called to obtain reward for the total claimable internal reward for the position.
+
+The reward is calculated in the following way(given a poolReward):
+
+```math
+\begin{aligned}
+\text{TotalRewardRatio}
+&=
+  \underbrace{\text{CRP}\bigl(\text{startHeight},\, h_1\bigr)}_{\text{CRP}(s,h_1)} \times r_1
+  \;+\;
+  \underbrace{\text{CRP}\bigl(h_1,\, h_2\bigr)}_{\text{CRP}(h_1,h_2)} \times r_2
+  \;+\;
+  \underbrace{\text{CRP}\bigl(h_2,\, \text{endHeight}\bigr)}_{\text{CRP}(h_2,e)} \times r_3,
+\\[6pt]
+\text{CRP}(a, b)
+&=
+  \text{CalcRaw}(b)
+  \;-\;
+  \text{CalcRaw}(a),
+\\[6pt]
+\text{CalcRaw}(h)
+&=
+  \begin{cases}
+    L(h) \;-\; U(h), 
+      & \text{if } \text{tick}(h) < \ell, \\[6pt]
+    U(h) \;-\; L(h), 
+      & \text{if } \text{tick}(h) \ge u, \\[6pt]
+    G(h) \;-\; \bigl(L(h) + U(h)\bigr), 
+      & \text{otherwise}.
```

Comment:
The functions $U(H)$, $L(h)$, and $G(h)$ are not defined. They appear to conceptually represent upper and lower bounds, But I think we need to include their separate definitions.